torch/csrc/jit/script/exit_transforms.cpp

```diff
@@ -76,11 +78,58 @@ struct ExitTransformer {
     false_val_ = graph_->insertConstant(false);
   };
 
-  void run() {
+  void run(Transform transform_kind) {
+    if (transform_kind == Transform::Returns) {
+      current_exit_kind_ = prim::ReturnStmt;
+      convertReturnOutputsToReturnStmts(graph_->block());
+    } else {
+      current_exit_kind_ = prim::LoopContinuation;
+      convertLoopOutputsToContinuations(graph_->block());
+    }
     transformExits(graph_->block());
   }
 
  private:
+  // The Logic for the exiy transform simplifies if the block outputs
+  // are converted to Exits before running, because you do not
```

Comment:
This comment is a bit unclear: is it a suggestion to improve (~simplify) the implementation in future, or is it an assumption we're already making? Also, it might help if we add a brief description of what function is expected to do.

Reply:
On second thought, it doesn't make it that much more simpler... I think I will remove it but i will leave that for a follow up.